lib/index.js

```diff
@@ -253,13 +244,22 @@ class ElectronDownloader {
   }
 
   moveFileToCache (filename, target, cb, onSuccess) {
-    debug('moving', filename, 'from', this.tmpdir, 'to', target)
-    fs.unlink(target, (err) => {
-      if (err != null && err.code !== 'ENOENT') return cb(err)
-      fs.move(path.join(this.tmpdir, filename), target, (err) => {
-        if (err) return cb(err)
+    const cache = this.cache
+    debug('moving', filename, 'from', cache, 'to', target)
+    fs.rename(path.join(cache, filename), target, (err) => {
+      if (err) {
+        fs.unlink(cache, cleanupError => {
+          try {
+            if (cleanupError) {
+              console.error(`Error deleting cache file: ${cleanupError.message}`)
+            }
+          } finally {
+            cb(err)
+          }
```

Comment:
User application can override `console.error`. e.g. Jest does it. As we must call user `cb` in any case`, it is more safe to use `try-catch` here. Otherwise we can get unresolved promise and so on.